TwitchClient/Endpoints/TwitchEndpoint.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace TwitchClient.Endpoints
+{
+    public abstract class TwitchEndpoint
+    {
+        public abstract string Format();
+
+        protected abstract IDictionary<string, string> CreateQueryParameters();
+
+        public Uri GetUri()
+        {
+            var apiBaseFormatted = this.Format();
+
+            var queryParameterDict = this.CreateQueryParameters();
+            var joined = queryParameterDict.Where(kv => kv.Value != null).Select(kv => string.Format("{0}={1}", kv.Key, kv.Value));
```

Comment:
I've always wanted an IEnumerable extension method that filters out null items.  Could you write this?  Maybe have a second one that takes in a selector function too for your case above.